### PR TITLE
Pass extract parameters to all readers and sources

### LINKED ISSUE #923: Integration of OEREBLex not multilingual
In the canton GR OEREBLex Supports multilingual calls in teh following form:

https://oereb-gr-preview.000.ch/api/geolinks/154.xml?locale=de
https://oereb-gr-preview.000.ch/api/geolinks/154.xml?locale=it
https://oereb-gr-preview.000.ch/api/geolinks/154.xml?locale=rm

The multilingual call of OEREBLex is not supported by pyramid_oereb. In the pyramid_oereb_standard.yml just one language can be defined for OEREBLex.

### PR DESCRIPTION
**Pull request by kdeininger**
Pass the `Parameter` object of the current request to all readers and sources to access its properties. This is needed to use the requested language when querying documents with the `OEREBlexSource`.

Closes #923

### CODE REVIEW

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046&bid=15146692)

File: pyramid_oereb/contrib/sources/document.py [hunk 1]
```diff
@@ -183,20 +193,21 @@ def _get_mapped_value(self, document, key, multilingual=False):
             if attribute:
                 value = getattr(document, attribute)
                 if value:
-                    return {self._language: value} if multilingual else value
+                    return {language: value} if language else value
         return None
 
-    def _get_document_title(self, document, current_file):
+    def _get_document_title(self, document, current_file, language):
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/sources/document.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

File: pyramid_oereb/standard/sources/legend.py [hunk 2]
```diff
@@ -6,11 +6,12 @@
 
 class DatabaseSource(BaseDatabaseSource, LegendBaseSource):
 
-    def read(self, **kwargs):
+    def read(self, params, **kwargs):
```
Comment (pfirpfel) [inline pyramid_oereb/standard/sources/legend.py]: Codacy Issue found: [Parameters differ from overridden 'read' method](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

File: pyramid_oereb/views/webservice.py [hunk 3]
```diff
@@ -452,14 +466,14 @@ def __parse_gnss__(self, gnss):
 
 
 class Parameter(object):
-    def __init__(self, flavour, format, with_geometry, images, identdn=None, number=None,
+    def __init__(self, format, flavour=None, with_geometry=False, images=False, identdn=None, number=None,
```
Comment (pfirpfel) [inline pyramid_oereb/views/webservice.py]: Codacy Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046&bid=15146692)

File: pyramid_oereb/standard/sources/legend.py (hunk 2 above)
Comment (pfirpfel) [inline pyramid_oereb/standard/sources/legend.py]: Codacy Issue found: [Parameters differ from overridden 'read' method](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046&bid=15146692)

File: pyramid_oereb/standard/sources/legend.py (hunk 2 above)
Comment (pfirpfel) [inline pyramid_oereb/standard/sources/legend.py]: Codacy Issue found: [Parameters differ from overridden 'read' method](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046&bid=15146692)

File: pyramid_oereb/standard/sources/legend.py (hunk 2 above)
Comment (pfirpfel) [inline pyramid_oereb/standard/sources/legend.py]: Codacy Issue found: [Parameters differ from overridden 'read' method](https://app.codacy.com/app/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046)

File: pyramid_oereb/views/webservice.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def _is_json(self):
             bool: True if requested format is JSON.
         """
         url = self._request.current_route_url().split('?')
-        return url[0].endswith('.json')
+        return self._request.matchdict.get('format', '').lower() == 'json' or url[0].endswith('.json')
```
Comment (jwkaltz) [inline pyramid_oereb/views/webservice.py]: I think this change is not related to the PR?

File: pyramid_oereb/views/webservice.py [hunk 5]
```diff
@@ -452,14 +466,14 @@ def __parse_gnss__(self, gnss):
 
 
 class Parameter(object):
-    def __init__(self, flavour, format, with_geometry, images, identdn=None, number=None,
-                 egrid=None, language=None, topics=None):
+    def __init__(self, response_format, flavour=None, with_geometry=False, images=False, identdn=None,
```
Comment (jwkaltz) [inline pyramid_oereb/views/webservice.py]: Why change the order of parameters?

Comment (kdeininger) [reply pyramid_oereb/views/webservice.py]: The `Parameter` object is now passed to the readers and sources. Some of them are also used for other requests, like `capabilities`, that only have the format parameter. So I had to turn some of the arguments into optional keyword arguments and therefore move them after the mandatory arguments.

I also renamed the format argument as `format` is a built-in function.

File: pyramid_oereb/views/webservice.py (hunk 4 above)
Comment (kdeininger) [reply pyramid_oereb/views/webservice.py]: I use this method for the creation of the `Parameter` object (e.g. line 162). I had to extend this method to respect also the format value of the request matchdict. The check of the URL ending only works for the old service URLs, where no format parameter has been defined in the specification.

File: pyramid_oereb/views/webservice.py (hunk 5 above)
Comment (jwkaltz) [reply pyramid_oereb/views/webservice.py]: Ah I see, thanks for the explanation

File: pyramid_oereb/views/webservice.py (hunk 4 above)
Comment (jwkaltz) [reply pyramid_oereb/views/webservice.py]: OK, thanks

**Comment by jwkaltz** [comment]
@kdeininger 
since this PR was approved, can you please check if you can resolve the conflicts and then merge?
I am asking because I think it makes sense to do a real release v1.5.2 (which could include this PR) soon, before merging some new features (from other PRs) which will require configuration changes and therefore should better be in a v1.5.3.

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4491046&bid=15146692)
